### PR TITLE
Trev8939/default challenge o auth

### PR DESCRIPTION
**Pull request by TADraeseke**
Thanks for your help with this @LDuncAndroid - Can you review when you have time? Thanks!

### CODE REVIEW

File: java/authenticate-with-oauth/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+## Use case
+
+Your app may need to access items that are only shared with authorized users. For example, your organization may host private data layers or feature services that are only accessible to verified users. You may also need to take advantage of premium ArcGIS Online services, such as geocoding or routing services, which require a named user login.
+
+## How to use the sample
+
+1. When you run the sample, the app will load a web map which contains premium content. You will be challenged for an ArcGIS Online login to view the private layers.
+1. Enter a user name and password for an ArcGIS Online named user account (such as your ArcGIS for Developers account).
+4. If you authenticate successfully, the traffic layer will display, otherwise the map will contain only the public basemap layer.
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/README.md]: Numbering here goes 1, 1, 4, 5

File: java/authenticate-with-oauth/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        <activity
+            android:name=".MainActivity"
+            android:label="@string/app_name">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+        <activity
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/src/main/AndroidManifest.xml]: Might be worth adding a comment here stating that this declaration of the `Activity` is required to display the OAuth login.

File: java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_main);
+
+    // get a reference to the map view
+    mMapView = findViewById(R.id.mapView);
+
+    try {
+      // set up an oauth config with url to portal, a client id and a re-direct url
+      // a custom client id for your app can be
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: Is this comment complete?

File: java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+      OAuthConfiguration oAuthConfiguration = new OAuthConfiguration(getString(R.string.portal_url),
+          getString(R.string.oauth_client_id),
+          BuildConfig.APPLICATION_ID + "://" + getString(R.string.oauth_redirect_host));
+
+      // setup AuthenticationManager to handle auth challenges
+      DefaultAuthenticationChallengeHandler defaultAuthenticationChallengeHandler = new DefaultAuthenticationChallengeHandler(
+          this);
+
+      // define this app's challenge handler as default
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: Maybe change to "use the `DefaultChallengeHandler` to handle authentication challenges" or similar

File: java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      AuthenticationManager.setAuthenticationChallengeHandler(defaultAuthenticationChallengeHandler);
+
+      // add an OAuth configuration
+      // NOTE: for this configuration to work, you must add an intent receiver to the app's manifest to handle starting
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: May be best to change to "you must add the `DefaultOAuthIntentReceiver` Activity to the app's manifest to handle starting a browser"

File: java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+      // a browser
+      AuthenticationManager.addOAuthConfiguration(oAuthConfiguration);
+
+      // load the
```
Comment (LDuncAndroid) [inline java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: * load the map

**Comment by TADraeseke** [comment]
Thanks @LDuncAndroid - Made those fixes!

File: java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java (hunk 6 above)
Comment (LDuncAndroid) [reply java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: @TADraeseke just this one to resolve

Comment (TADraeseke) [reply java/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.java]: Oops thanks, fixed that now too @LDuncAndroid

**Comment by LDuncAndroid** [comment]
Great, LGTM 👍

**Pull request merged**